lib/src/zenohUri.cpp

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright (c) 2024 General Motors GTO LLC
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ * SPDX-FileType: SOURCE
+ * SPDX-FileCopyrightText: 2024 General Motors GTO LLC
+ * SPDX-License-Identifier: Apache-2.0
+ */
+ 
+#include <iomanip>
+#include <sstream>
+#include <spdlog/spdlog.h>
+#include <up-cpp/uri/serializer/MicroUriSerializer.h>
+#include "up-client-zenoh-cpp/session/zenohSessionManager.h"
+
+namespace uprotocol::uri {
+
+namespace {
+void hexlify(auto& topic, const uint8_t byte) {
+    topic << std::hex << std::setw(2) << std::setfill('0');
+    topic << static_cast<int>(byte);
+}
+
+void hexlify(auto& topic, const auto &&start, const auto &&end) {
+    for (auto i = start; i < end; ++i) {
+        hexlify(topic, *i);
+    }
+}
+
+} // anonymous namespace
+
+std::string toZenohKeyString(const uprotocol::v1::UUri &u_uri) {
```

Comment:
do you have a check if the provided URI is long URI? do you return an error?

Reply:
If it is a long URI, then the micro URI serializer will fail and return an empty string. The current design does not have a way to propagate errors other than returning empty strings.